demos/meltdown_ss.cc

```diff
@@ -58,6 +58,7 @@ const char *public_data = "Hello, world!";
 static void setup_segment(int index, const char *base, bool present) {
   // See: Intel SDM volume 3a "3.4.5 Segment Descriptors".
   struct user_desc table_entry;
+  memset(&table_entry, 0, sizeof(struct user_desc));
```

Comment:
This is only to avoid undefined behavior or were there other motivations for avoiding unitialized memory?

Reply:
Only to avoid undefined behavior. I detected it by valgrind.